src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -329,6 +329,9 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
         writeChar('\n', statement_buf);
         String statement = statement_buf.str();
 
+        /// Needed to make database creation retriable if it fails after the file is created
+        fs::remove(metadata_file_tmp_path);
```

Comment:
Just in case - we can check ctime/mtime to make sure that it was created recently

Reply:
This is a `.tmp` file that we rename later during database creation. Even if it was not created recently, it seems like it's still safe to reuse it.